src/backends/torch/torchlib.cc

```diff
@@ -211,37 +249,48 @@ namespace dd
       torch::load(_classif, model._weights, _device);
     if (!model._proto.empty())
       {
-        _native = std::make_shared<CaffeToTorch>(model._proto);
+        _graph = std::make_shared<CaffeToTorch>(model._proto);
         if (!model._traced.empty())
-          torch::load(_native, model._traced, _device);
+          torch::load(_graph, model._traced, _device);
+      }
+    if (!model._native.empty())
+      {
+        std::shared_ptr<NativeModule> m;
+        torch::load(m, model._native);
+        _native = m;
       }
   }
 
   void TorchModule::eval()
   {
-    if (_native)
-      _native->eval();
+    if (_graph)
+      _graph->eval();
     if (_traced)
       _traced->eval();
     if (_classif)
       _classif->eval();
+    if (_native)
+      _native->eval();
   }
 
   void TorchModule::train()
   {
-    if (_native)
-      _native->train();
+    if (_graph)
+      _graph->train();
     if (_traced)
       _traced->train();
     if (_classif)
       _classif->train();
+    if (_native)
+      _native->eval();
```

Comment:
Should be `train()` ?

Reply:
it should. nice catch!

Reply:
was it even training ? ;)

Reply:
mmmm

Reply:
train() is passed to all module and they do what they want with it, ie switch on/off dropout batchnorm and so on, most models do not do anything with it